lib/rdf/repository.rb

```diff
@@ -195,22 +197,18 @@ def transaction(graph_name = nil, &block)
     # @param  [RDF::Resource] graph_name
     # @return [RDF::Transaction]
     # @since  0.3.0
-    def begin_transaction(graph_name)
-      RDF::Transaction.new(graph: graph_name)
+    def begin_transaction(mutable: false)
+      @tx_class.new(self, mutable: mutable)
     end
 
     ##
     # Rolls back the given transaction.
     #
-    # Subclasses implementing transaction-capable storage adapters may wish
-    # to override this method in order to roll back the given transaction in
-    # the underlying storage.
-    #
     # @param  [RDF::Transaction] tx
     # @return [void] ignored
     # @since  0.3.0
     def rollback_transaction(tx)
```

Comment:
I'm reminded (again) that `#freeze` doesn't freeze instance variables of the frozen object.

Considering just adding a `@rolledback` flag, and raising an error on `#execute`. If it's true. (could do that with `#frozen?`, too, but it feels like a semantic overload)